### DISCUSSION TITLE
The issue on EfiSimpleNetwork

### DISCUSSION THREAD

**Discussion by amdhugo** [ipxe/ipxe#281] [Q&A]
We hit an issue and the BIOS guy replies us as below.
The root cause is the iPxe driver installed protocol "EfiSimpleNetwork" will be free, if the image return errors. and the MnpReceivePacket is also running as a timer event.This change is only a W/A for this issue. the fixed should be the iPxe driver to uninstall the EfiSimpleNetwork protocol, and stop the MnpReceivePacket() polling if error found.

I post this information here in order to ask if iPXE agree this issue and plan to fix it?

**Comment by mcb30** [comment]
iPXE **does** uninstall the EfiSimpleNetwork protocol.  The MNP driver should react to this by stopping polling the SNP device.  If the MNP driver does not do this then there is a bug in the MNP driver.

As of commit https://github.com/ipxe/ipxe/commit/5b41b9a80 (from October 2020) iPXE will work around this kind of bug in the MNP driver (or other consumers of protocols published by iPXE): if the call to `UninstallMultipleProtocolInterfaces` reports an error due to a buggy BIOS driver then iPXE will deliberately leave the protocol present but disabled, leaking the memory since it is not safe to free the protocol.

I am marking this conversation as locked since it is not really suitable for the discussion forum.  Please reraise as an issue via https://github.com/ipxe/ipxe/issues if your BIOS guy is unable to fix the bug in the BIOS implementation of the MNP protocol.
